From fc381b5938f7235401286d4821f03d2b05d4876a Mon Sep 17 00:00:00 2001
From: Emilio Pozuelo Monfort <pochu27@gmail.com>
Date: Sat, 27 Feb 2010 15:32:23 +0100
Subject: [PATCH] Don't quit zenity when the input stream is closed

When using --listen for a notification, zenity quits when the stream
ends. This makes it impossible to read commands from a pipe as in
`echo icon:info | zenity --notification --listen'.

https://bugzilla.gnome.org/show_bug.cgi?id=525596
---
 src/notification.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

Index: zenity-3.2.0/src/notification.c
===================================================================
--- zenity-3.2.0.orig/src/notification.c	2011-09-27 00:42:16.000000000 +0200
+++ zenity-3.2.0/src/notification.c	2011-09-28 15:39:04.809489648 +0200
@@ -167,8 +167,6 @@
 
   if ((condition & G_IO_HUP) != 0) {
     g_io_channel_shutdown (channel, TRUE, NULL);
-    zen_data->exit_code = zenity_util_return_exit_code (ZENITY_OK);
-    gtk_main_quit ();
     return FALSE;
   }
 
